core/domain/blog_services.py

```diff
@@ -56,7 +74,33 @@ def get_blog_post_from_model(blog_post_model):
         blog_post_model.published_on)
 
 
-def get_blog_post_by_id(blog_post_id, strict=True):
+@overload
+def get_blog_post_by_id(
+    blog_post_id: str
+) -> blog_domain.BlogPost: ...
+
+
+@overload
+def get_blog_post_by_id(
+    blog_post_id: str, strict: Literal[True]
+) -> blog_domain.BlogPost: ...
+
+
+@overload
+def get_blog_post_by_id(
+    blog_post_id: str, strict: Literal[False]
+) -> Optional[blog_domain.BlogPost]: ...
+
+
+@overload
+def get_blog_post_by_id(
+    blog_post_id: str, strict: bool = False
+) -> Optional[blog_domain.BlogPost]: ...
+
+
+def get_blog_post_by_id(
+    blog_post_id: str, strict: bool = True
+) -> Optional[blog_domain.BlogPost]:
```

Comment:
Should we enforce the `strict` to always be set to some value? Then we would be able to set the typing so that when `strict` is `True` then it cannot be `Optional`?
```suggestion
def get_blog_post_by_id(
    blog_post_id: str, strict: bool = True
) -> Optional[blog_domain.BlogPost]:
```

Reply:
I think there are cases in the codebase where we are intentionally passing `strict = False`, If we set `strict = True` just to remove `Optional` in every case then I'm not sure that it's a good idea.

Reply:
Wait I don't understand you comment. Can you try to rephrase?

Reply:
Ahh sorry!!!,  actually I misinterpreted the review comment. but now i have done the suggested changes.